### PR TITLE
ci: define commit scope values

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Which issue(s) does this pull-request address?

Closes: #49 

### Description

Based on the discussion in #49.  The current list of scopes are in https://github.com/i-am-bee/bee-agent-framework/issues/49#issuecomment-2400490099

### Checklist

- [x] I have read the [contributor guide](https://github.com/i-am-bee/bee-agent-framework/blob/main/CONTRIBUTING.md)
- [x] Linting passes: `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: `yarn format` or `yarn format:fix`
- [x] Unit tests pass: `yarn test:unit`
- [ ] E2E tests pass: `yarn test:e2e`
- [ ] Tests are included 
- [x] Documentation is changed or added
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +22,24 @@
         2,
         "always",
         "Signed-off-by"
+      ],
+      "scope-enum": [
+        2,
```
Comment (markstur) [inline package.json:27]: Shouldn't this be a warning instead of an error?   I mean it says "optional" in the above CONTRIBUTING.md explainer.

Comment (akihikokuroda) [reply package.json:27]: In the scope-enum section in https://commitlint.js.org/reference/rules.html has a note

> This rule always passes if no scopes are provided in the message or the value > is an empty array.

It seems that the scope is always optional.

Comment (markstur) [reply package.json:27]: > In the scope-enum section in https://commitlint.js.org/reference/rules.html has a note
> 
> > This rule always passes if no scopes are provided in the message or the value > is an empty array.
> 
> It seems that the scope is always optional.

Sounds good to me.  I tried to figure out always|never|optional but missed the scope-enum detail (and there is no optional).

**Comment by akihikokuroda** [comment]
Hi!  This is a friendly remainder.  Could this be reviewed?  Thanks!

**Comment by Tomas2D** [comment]
I will look at the beginning of next week (@akihikokuroda). Thank you.

File: package.json [hunk 2]
```diff
@@ -22,6 +22,24 @@
...
+          "agents",
+          "llms",
+          "tools",
+          "cache",
+          "emitter",
+          "internals",
+          "logger",
+          "memory",
+          "serializer",
+          "repo-infra",
```
Comment (Tomas2D) [inline package.json:40]: I would just use `infra`

File: package.json [hunk 3]
```diff
@@ -22,6 +22,24 @@
         2,
         "always",
         "Signed-off-by"
+      ],
+      "scope-enum": [
+        2,
+        "always",
+        [
```
Comment (Tomas2D) [inline package.json:29]: add `instrumentation`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -78,7 +78,7 @@ WATSONX_DEPLOYMENT_ID=""  # optional
 ```
 <type>(<scope>): <subject>
 Type: feat, fix, chore, docs, style, refactor, perf, test, etc.
-Scope: The area of the codebase your changes affect (optional).
+Scope: The area of the codebase your changes affect (optional).    The allowed values are: adapters, agents, llms, tools, cache, emitter, internals, logger, memory, serializer, repo-infra, deps.
```
Comment (Tomas2D) [inline CONTRIBUTING.md:81]: I would add a link to `package.json` where one can find the current list.

**Review by Tomas2D — COMMENTED** [review]
Just a few notes. Then, we can merge. Thank you, and sorry for waiting.

**Comment by akihikokuroda** [comment]
@Tomas2D Thanks for review.  I addressed all your comments.

**Review by Tomas2D — APPROVED** [review]
Thank you for the contribution 🚀

**Pull request merged**